st2api/st2api/controllers/resource.py

```diff
@@ -246,6 +246,25 @@ def _get_from_model_kwargs_for_request(self, request):
         """
         return self.from_model_kwargs
 
+    def _get_one_by_scope_and_name(self, scope, name, from_model_kwargs=None):
+        """
+        Retrieve an item given scope and name. Only KeyValuePair now has concept of 'scope'.
+
+        :param scope: Scope the key belongs to.
+        :type scope: ``str``
+
+        :param name: Name of the key.
+        :type name: ``str``
+        """
+        instance = self.access.get_by_scope_and_name(scope=scope, name=name)
+        if not instance:
+            return instance
```

Comment:
>Also seems like this is kinda key value pair controller specific so maybe it should live in base key value pair controller or similar?

I had it in KeyValuePairController and manas said move it to some base layer. Geez. Ignoring.